static/src/js/components/PermissionForm/PermissionForm.jsx

```diff
@@ -0,0 +1,647 @@
+  }
+
+  // Validate groupPermissions
+  if (isEmpty(formData.groupPermissions) || formData.groupPermissions.length === 0) {
+    errors.groupPermissions.addError('At least one group permission must be specified')
+  }
+
+  return errors
+}
+
+const PermissionForm = () => {
+  const {
+    draft,
+    originalDraft,
+    setDraft,
+    setOriginalDraft,
+    providerId,
+    setProviderId
+  } = useAppContext()
+
+  const navigate = useNavigate()
+
+  const { addNotification } = useNotificationsContext()
+
+  const { conceptId = 'new' } = useParams()
+
+  const [focusField, setFocusField] = useState(null)
+  const [uiSchema, setUiSchema] = useState(collectionPermissionUiSchema)
+
+  const [chooseProviderModalOpen, setChooseProviderModalOpen] = useState(false)
+
+  const [createAclMutation] = useMutation(CREATE_ACL)
+  const [updateAclMutation] = useMutation(UPDATE_ACL)
+
+  const fields = {
+    keywordPicker: KeywordPicker,
+    TitleField: CustomTitleField,
+    CollectionSelector: CollectionSelectorPage,
+    GroupPermissionSelect,
+    layout: GridLayout
+  }
+
+  const widgets = {
+    TextWidget: CustomTextWidget,
+    SelectWidget: CustomSelectWidget,
+    DateTimeWidget: CustomDateTimeWidget
+  }
+
+  const templates = {
+    ArrayFieldTemplate: CustomArrayFieldTemplate,
+    FieldTemplate: CustomFieldTemplate,
+    TitleField: CustomTitleFieldTemplate
+  }
+
+  const { data } = useSuspenseQuery(GET_COLLECTION_FOR_PERMISSION_FORM, {
+    skip: conceptId === 'new',
+    variables: {
+      conceptId
+    }
+  })
+
+  useEffect(() => {
+    if (conceptId === 'new') {
+      setDraft({})
+      setOriginalDraft({})
+    }
+  }, [conceptId])
+
+  const showGranuleFields = (formData) => {
+    if (formData.accessPermission?.granule) {
+      const newUiSchema = {
+        ...collectionPermissionUiSchema,
+        accessConstraintFilter: {
+          ...collectionPermissionUiSchema.accessConstraintFilter,
+          granuleAccessConstraint: {
+            ...collectionPermissionUiSchema.accessConstraintFilter.granuleAccessConstraint,
+            'ui:disabled': false
+          }
+        },
+        temporalConstraintFilter: {
+          ...collectionPermissionUiSchema.temporalConstraintFilter,
+          granuleTemporalConstraint: {
+            ...collectionPermissionUiSchema.temporalConstraintFilter.granuleTemporalConstraint,
+            'ui:disabled': false
+          }
+        }
+      }
+      setUiSchema(newUiSchema)
+    } else {
+      const newUiSchema = {
+        ...collectionPermissionUiSchema,
+        accessConstraintFilter: {
+          ...collectionPermissionUiSchema.accessConstraintFilter,
+          granuleAccessConstraint: {
+            ...collectionPermissionUiSchema.accessConstraintFilter.granuleAccessConstraint,
+            'ui:disabled': true
+          }
+        }
+      }
+      setUiSchema(newUiSchema)
+    }
+  }
+
+  useEffect(() => {
```

Comment:
This useEffect is 149 lines long with no comments explaining anything ;) Describe the purpose of the useEffect, and comment logic throughout